src/attributes/codegen.md

```diff
@@ -66,20 +65,42 @@ It is [undefined behavior] to call a function that is compiled with a feature
 that is not supported on the current platform the code is running on, *except*
 if the platform explicitly documents this to be safe.
 
-Functions marked with `target_feature` are not inlined into a context that
-does not support the given features. The `#[inline(always)]` attribute may not
-be used with a `target_feature` attribute.
+For this reason, a function marked with `target_feature` is unsafe, except in
+a context that supports the given features. For example:
+
+```rust
+# #[target_feature(enable = "avx2")]
+# fn foo_avx2() {}
+
+fn bar() {
+    // Calling `foo_avx2` here is unsafe, as we must ensure
+    // that AVX is available first.
+    unsafe {
+        foo_avx2();
+    }
+}
+
+#[target_feature(enable = "avx2")]
+fn bar_avx2() {
+    // Calling `foo_avx2` here is safe.
+    foo_avx2();
+    || foo_avx2();
+}
+```
+
+Like unsafe functions, functions marked with `target_feature` cannot be
+assigned to a safe function pointer and do not implement `FnOnce`.
```

Comment:
Can this clarify that it *can* be used with an unsafe function pointer?  Like another sentence along the lines of "They may be assigned to an `unsafe` function pointer, for example:"

And also include an example, perhaps something like:

```rust
#[target_feature(enable = "avx2")]
fn fn_ptr_example() {}

// `unsafe` is required here
static MY_FN_PTR: unsafe fn () -> () = fn_ptr_example;
```